### PR TITLE
Fix warning in keras base_layer_utils.py

### PR DESCRIPTION
**Pull request by yongtang**
While debugging 24741 I noticed the following warning:
```
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer_utils.py:125: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
```

This fix update the base_layer_utils.py to remove the warning.
Note dtype is passed in the subsequent calls already so there is no need to
call with dtype in __init__ in this case.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by fchollet — DISMISSED** [review]
LGTM, thanks for the PR

**Comment by yongtang** [comment]
Thanks @fchollet for the review. The PR has been rebased and updated to fix the merge conflict.

**Comment by hgadig** [comment]
@fchollet  Could you please approve this if it looks good. Once it is approved, then I can proceed further.

**Review by fchollet — APPROVED** [review]

**Pull request merged**
